File: tmp/capture_screenshots.py
import os
import sys
import time
from playwright.sync_api import sync_playwright

output_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), 'review-screenshots'))
os.makedirs(output_dir, exist_ok=True)

viewports = [
    {'name': 'mobile', 'width': 390, 'height': 844},
    {'name': 'desktop', 'width': 1440, 'height': 900}
]

routes = ['/login', '/']

screenshots_taken = []

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    
    for vp in viewports:
        context = browser.new_context(
            viewport={'width': vp['width'], 'height': vp['height']},
            device_scale_factor=1
        )
        page = context.new_page()
        
        # Capture console errors
        console_errors = []
        page.on("console", lambda msg: console_errors.append(msg.text) if msg.type == "error" else None)
        
        for route in routes:
            url = f"http://localhost:4200{route}"
            print(f"Navigating to {url} ({vp['name']} {vp['width']}x{vp['height']})...")
            
            page.goto(url)
            page.wait_for_load_state("networkidle")
            time.sleep(1) # Extra stability
            
            clean_route = route.replace('/', '_').strip('_') or 'home'
            filename = f"{clean_route}_{vp['name']}_{vp['width']}x{vp['height']}.png"
            filepath = os.path.join(output_dir, filename)
            
            page.screenshot(path=filepath, full_page=False)
            screenshots_taken.append(filepath)
            print(f"Captured: {filepath}")
            
            if console_errors:
                print(f"Console errors on {route}: {console_errors}")
                
        context.close()
        
    browser.close()

print("\nAll screenshots captured successfully:")
for path in screenshots_taken:
    print(path)

File: tmp/capture_screenshots.js
const { chromium } = require('playwright-core');
const fs = require('fs');
const path = require('path');

const outputDir = path.resolve(__dirname, 'review-screenshots');
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

const viewports = [
  { name: 'mobile', width: 390, height: 844 },
  { name: 'desktop', width: 1440, height: 900 }
];

const routes = ['/login', '/'];

async function main() {
  let executablePath;
  const possiblePaths = [
    'C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe',
    'C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe',
    'C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe',
    'C:\\Program Files\\Microsoft\\Edge\\Application\\msedge.exe'
  ];
  for (const p of possiblePaths) {
    if (fs.existsSync(p)) {
      executablePath = p;
      break;
    }
  }

  console.log('Using browser executable:', executablePath);
  const browser = await chromium.launch({ executablePath, headless: true });

  const screenshotsTaken = [];

  for (const vp of viewports) {
    const context = await browser.newContext({
      viewport: { width: vp.width, height: vp.height },
      deviceScaleFactor: 1
    });
    const page = await context.newPage();

    page.on('console', msg => {
      if (msg.type() === 'error') {
        console.error(`[Console Error on ${page.url()}]: ${msg.text()}`);
      }
    });

    for (const route of routes) {
      const url = `http://localhost:4200${route}`;
      console.log(`Navigating to ${url} (${vp.name} ${vp.width}x${vp.height})...`);

      if (route === '/') {
        // Set auth token for home route
        await page.goto('http://localhost:4200/login', { waitUntil: 'domcontentloaded' });
        await page.evaluate(() => localStorage.setItem('quiz_access_token', 'demo_access_token_123'));
      } else if (route === '/login') {
        // Clear auth token for login route
        await page.goto('http://localhost:4200/login', { waitUntil: 'domcontentloaded' });
        await page.evaluate(() => localStorage.removeItem('quiz_access_token'));
      }

      await page.goto(url, { waitUntil: 'domcontentloaded' });
      await page.waitForTimeout(1200);

      const cleanRoute = route.replace(/\//g, '_').replace(/^_/, '') || 'home';
      const filename = `${cleanRoute}_${vp.name}_${vp.width}x${vp.height}.png`;
      const filepath = path.join(outputDir, filename);

      await page.screenshot({ path: filepath, fullPage: false });
      screenshotsTaken.push(filepath);
      console.log(`Captured: ${filepath}`);
    }

    await context.close();
  }

  await browser.close();

  console.log('\nAll screenshots captured successfully:');
  screenshotsTaken.forEach(s => console.log(s));
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});

